.github/workflows/ci-build.yaml

```diff
@@ -0,0 +1,92 @@
+name: CI
+
+on:
+  # Trigger the workflow on push or pull request,
+  # only on the master or jax branches
+  push:
+    branches:
+      - master
+      - jax
+  pull_request:
+    branches:
+      - master
+      - jax
+
+jobs:
+  lint_and_typecheck:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.7]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Get pip cache dir
+      id: pip-cache
+      run: |
+        python -m pip install --upgrade pip
+        echo "::set-output name=dir::$(pip cache dir)"
+    - name: pip cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ steps.pip-cache.outputs.dir }}
+        key: ${{ runner.os }}-pip-${{ hashFiles('**/setup.py', '**/test-requirements.txt') }}
```

Comment:
Rather than placing the test requirements in a separate file, can we keep them in setup.py along with other dependencies?

e.g. (untested)

```
setup(
   ...
   install_requires=REQUIRED_PACKACES,
   extras_require={'testing': ['pytest', ...]},
   ...
}
```

I think we could then do

```
pip install -e '.[testing]'
python -m pytest
```

Again, untested. :)

It would also be great if you could update the README to add installing packages required to run the tests!